automl/automl-script/script.whizzml

```diff
@@ -151,12 +171,31 @@
 
 ;; Obtains the most important fields from a dataset
 ;; using recursive feature elimination
-(define (compute-important-fields train test)
+(define (compute-important-fields train test shallow)
   (let (balance (config "balance-objective")
-        rfe-output (recursive-feature-elimination train 1 "" test "" balance)
+        train (wait (sampled-ds-maybe train shallow))
+        num-fields (count (keys (resource-fields train)))
+        min-fields-shallow 30
+        min-fields 50
+        rfe-output (if shallow
+                     ;; If shallow-search, fixed number
+                     ;; of fields  after rfe, 30
+                     ;; No evaluations
+                     (rfe train min-fields-shallow "" ""   "" balance)
+                     ;; We want, at least 50 fields. If the total
+                     ;; number of fields is lower, we will skip
+                     ;; feature selection stage
+                     (rfe train min-fields "" test "" balance))
         evaluations (rfe-output "evaluations" {})
-        num-features (rfe-best-num-features evaluations))
-    (rfe-output-fields train evaluations num-features)))
+        num-features (if shallow
+                       (if (> num-fields min-fields-shallow)
```

Comment:
isn't it 
(min num-fields min-fields-shallow)

Reply:
Yes, thanks!